packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts

```diff
@@ -411,21 +411,31 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
         allowMinify &&
         (buildOptions.platform == 'server'
           ? {
-            ecma: terserEcma,
-            global_defs: angularGlobalDefinitions,
-            keep_fnames: true,
-          }
+              ecma: terserEcma,
+              global_defs: angularGlobalDefinitions,
+              keep_fnames: true,
+            }
           : {
-            ecma: terserEcma,
-            pure_getters: buildOptions.buildOptimizer,
-            // PURE comments work best with 3 passes.
-            // See https://github.com/webpack/webpack/issues/2899#issuecomment-317425926.
-            passes: buildOptions.buildOptimizer ? 3 : 1,
-            global_defs: angularGlobalDefinitions,
-          }),
+              ecma: terserEcma,
+              pure_getters: buildOptions.buildOptimizer,
+              // PURE comments work best with 3 passes.
+              // See https://github.com/webpack/webpack/issues/2899#issuecomment-317425926.
+              passes: buildOptions.buildOptimizer ? 3 : 1,
+              global_defs: angularGlobalDefinitions,
+            }),
       // We also want to avoid mangling on server.
       // Name mangling is handled within the browser builder
-      mangle: allowMangle && buildOptions.platform !== 'server' && !differentialLoadingMode,
+      mangle: allowMangle &&
+        buildOptions.platform !== 'server' &&
+        !differentialLoadingMode && {
+          properties: {
+            regex: /^(?:ɵfac|ɵcmp|ɵinj|ɵmod)$/,
+          },
+        },
+      nameCache: {
+        vars: { props: {} },
+        props: { props: { $ɵmod: 'm', $ɵinj: 'i', $ɵfac: 'f', $ɵcmp: 'c' } },
```

Comment:
What happens if a component (say) already has a property called `c`?

Reply:
These changes would also brea analytics build stats

Reply:
The existing static property gets overwritten by the Angular property.  This is the same as current behavior (although less likely with the more exotic name) and is really a consequence of using string property names.  To fix this we could consider using symbol names instead.  They work well as a metadata store.

Reply:
Yeah but `ɵcmp` is much less likely to be used on a component than `c`...

Reply:
I did mention that :)

Alternatively, we could try running terser serially and let it pick a unique name; although at the cost of build time.

Reply:
I can see the appeal but should we really try to reinvent compression? Surely, everyone delivers static assets compressed by gzip or brotli or...

Reply:
I see, thank you for expanding. (: